### ISSUE TITLE
I am making an docker image for this project

### ISSUE THREAD

**Issue by boyang9602** [zpw6106/tloam#11]
Update

Check: https://github.com/boyang9602/tloam/tree/docker
Still in progress due to some runtime errors.  

--------------
Hi,

I am making a docker image for this project https://github.com/boyang9602/TLOAM-docker. I met the same problems as https://github.com/zhoupengwei/tloam/issues/8 and https://github.com/zhoupengwei/tloam/issues/9 and I do not have a solution right now. 

Now, all the dependencies are correctly installed—anyone interested please feel free to use and contribute to it. Thank you very much!

**Comment by boyang9602** [comment]
I think I found the cause of the building error

TLOAM uses Open3D v0.12 and implemented PointCloud2 which extended from open3d::geometry::Geometry3D. There is a difference of `GetOrientedBoundingBox` function in Open3D v0.12 and v0.15 for Geometry3D

In [v0.12](https://github.com/isl-org/Open3D/blob/313315d9790c36e22bb5bb034e9c7d7f470cdf73/cpp/open3d/geometry/Geometry3D.h#L68)

```c++
    /// Returns an oriented bounding box of the geometry.
    virtual OrientedBoundingBox GetOrientedBoundingBox() const = 0;
```

while in [v0.15](https://github.com/isl-org/Open3D/blob/1aa25b9abff7dfa25ec2519b86d391ee34fa2f0d/cpp/open3d/geometry/Geometry3D.h#L75-L76)

```c++
    virtual OrientedBoundingBox GetOrientedBoundingBox(
            bool robust = false) const = 0;
```

So if we use Open3D v0.15+, PointCloud2 will have an un-implemented virtual function.

**Comment by boyang9602** [comment]
I made a fix about the compatibility with open3d v0.15 https://github.com/boyang9602/tloam/tree/o3dv0.15

**Comment by zhoupengwei** [comment]
@boyang9602 Thank you for your attention for my work, these problem have been fixed in the new [PR](https://github.com/zhoupengwei/tloam/pull/12). If you have verified that the provided Docker image works well, please proceed to create a new PR for my project. Thank you for your contribution.

**Comment by gzyabc** [comment]
@boyang9602 ， @zhoupengwei 。Hello I was able to compile successfully, but I reported a new error while launching the launch file.How can I solve it
![pic1](https://github.com/user-attachments/assets/f06c7214-5c5b-4365-9cd9-c322873e191c)

**Comment by gzyabc** [comment]
![pic2](https://github.com/user-attachments/assets/3ec6c708-1c72-4efa-bc5b-a0cc98bef3c5)
I don't know why this is, bu guys can you help me？

**Comment by boyang9602** [comment]
Hi @gzyabc , I have the same problem. I do not have a solution right now.

**Comment by gzyabc** [comment]
@boyang9602 So you just compile, but the code doesn't actually run?

**Comment by boyang9602** [comment]
No, it could not run properly. I had the same error messages: https://github.com/zhoupengwei/tloam/pull/13
